src/new.spec.js

```diff
@@ -234,4 +234,59 @@ describe("withStore", () => {
       });
     });
   });
+
+  describe("effects", () => {
+    it("receives the passed arguments", () => {
+      const args = ["bar", "baz"];
+      const { effects } = makeTestInstance({
+        effects: {
+          foo: (...rest) => {
+            expect(rest).toEqual(args);
+          },
+        },
+      });
+      return effects.foo(...args);
+    });
+
+    it("are called with read-only effects and props and resetState and writable state in context", () => {
+      const state = { foo: "bar" };
+      const { effects, getParentProps } = makeTestInstance({
+        initialState: () => state,
+        effects: {
+          foo() {
+            assert(isReadOnly(this));
+
+            expect(typeof this.resetState).toBe("function");
+
+            assert(isReadOnly(this.effects));
+            expect(this.effects).toBe(effects);
+
+            assert(isReadOnly(this.props));
+            expect(this.props).toBe(getParentProps());
+
+            expect(state.foo).toBe(this.state.foo);
+            this.state.foo = "baz";
+            expect(this.state.foo).toBe("baz");
+          },
```

Comment:
Like we did for the first param of the render function, check that `effects` and `state` are what we expected.